### DISCUSSION TITLE
Cypress v10+ Client Code Instrumentation at Runtime

### DISCUSSION THREAD

**Discussion by clazette** [cypress-io/cypress#24349] [CI setup]
I would like to instrument client code, in Cypress, on-the-fly vs during the build process. Has anyone had success with that approach?

**Comment by thiswallz** [comment]
hi @clazette , for this kind of things you should use something a bit advance to run things dynamically on the client side (like tricking the building time), with a module loader such as SystemJS, check this: https://github.com/systemjs/systemjs.

what it does is that import your module dynamically in run-time, so i guess you can change the code behind and re-load it in the client side (without any server interaction whatsoever) websites such as code sandbox or codepen use similar strategies, otherwise would be a crazy usage of infrastructure for free.

here is an example, this is doing some client side load: https://www.freecodecamp.org/learn/front-end-development-libraries/redux/use-const-for-action-types
